### DISCUSSION TITLE
[BUG] Helm update 1.5.1 -> 1.5.5 dry-run nil pointer

### DISCUSSION THREAD

**Discussion by eqinox76** [longhorn/longhorn#8834] [General]
## Describe the bug

I am trying to update from longhorn 1.5.1 to 1.5.5 by using helm.

## To Reproduce

```
$ helm -n longhorn list
NAME            NAMESPACE       REVISION        UPDATED                                         STATUS          CHART           APP VERSION
longhorn        longhorn        1               2023-09-27 13:05:36.839412456 +0200 CEST        deployed        longhorn-1.5.1  v1.5.1     
$ helm -n longhorn get values longhorn
USER-SUPPLIED VALUES:
defaultSettings:
  backupTarget: s3://my-backup@EUROPE-WEST3/
  backupTargetCredentialSecret: my-backups
$ helm upgrade longhorn longhorn/longhorn --namespace longhorn --version 1.5.5 --dry-run=server --reuse-values
Error: UPGRADE FAILED: template: longhorn/templates/servicemonitor.yaml:1:15: executing "longhorn/templates/servicemonitor.yaml" at <.Values.metrics.serviceMonitor.enabled>: nil pointer evaluating interface {}.serviceMonitor
```

## Expected behavior

Helm shows what would change when running the upgrade.

## Support bundle for troubleshooting

Its a helm deployment issue.

## Environment

 - Longhorn version: 1.5.1
 - Impacted volume (PV): 
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: RKE2

## Additional context

As this happens in our productive system i did not yet run the command without `--dry-run`.

**Comment by innobead** [comment]
This new setting was introduced to 1.6 and backported to 1.5.4. https://github.com/longhorn/longhorn/issues/7041

You need to set this new value instead of only using `--reuse-values`. 
>       --reuse-values                               when upgrading, reuse the last release's values and merge in any overrides from the command line via --set and -f. If '--reset-values' is specified, this is ignored

**Comment by eqinox76** [comment]
Thanks for the explanation!
```
helm upgrade longhorn longhorn/longhorn --namespace longhorn --version 1.5.5 --reuse-values --set metrics.serviceMonitor.enabled=false,helmPreUpgradeCheckerJob.enabled=true
```
is working as expected.
